lib/subscriber_list/hash_digest.rb

```diff
@@ -0,0 +1,27 @@
+require 'digest'
+
+class SubscriberList
+  class HashDigest
+    def self.generate(hash = {})
```

Comment:
This class has ended up with a somewhat unusual interface with this and sort_hash quasi private method. I'd suggest you actually initialize an object to perform this conversion and then have a self.generate method that is like `new(hash).generate`

Also minor but I assume we don't really need the default argument of `{}` since we'd not expect someone to call this without an argument.